tests/test_init.py

```
@@ -249,3 +252,71 @@ async def test_save_capabilities_writes_library_dict_directly(hass):
+    with patch("custom_components.givenergy_local.async_expose_entity") as expose_mock:
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_EXPOSE_RECOMMENDED_ENTITIES,
+            {
+                "device_id": inverter_device.id,
+                "assistants": ["conversation", "cloud.alexa"],
+            },
+            blocking=True,
+        )
+
+    # Two assistants × N entities → 2N total exposure calls.
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)